src/OrbitGl/TimelineUiTest.cpp

```diff
@@ -72,24 +88,36 @@ class TimelineUiTest : public TimelineUi {
   MockBatcher mock_batcher_;
   MockTextRenderer mock_text_renderer_;
   PrimitiveAssembler primitive_assembler_;
-  TimeGraphLayout layout_;
-  Viewport viewport_;
-  MockTimelineInfo mock_timeline_info;
+  Viewport* viewport_;
+  MockTimelineInfo* mock_timeline_info_;
 };
 
-TEST(TimelineUI, UpdatePrimitivesBigScreen) {
-  // TODO(b/218311326): In some cases, small screens and even medium screens (Width < 700) failed in
-  // the condition between #labels and #major_ticks, since we are drawing major_ticks anyways when
-  // labels intersect.
-  const int kBigScreenWidth = 2000;
-  TimelineUiTest timeline_ui_test(kBigScreenWidth);
+static void TestUpdatePrimitivesWithSeveralRanges(int world_width) {
+  MockTimelineInfo mock_timeline_info;
+  Viewport viewport(0, 0);
+  TimeGraphLayout layout;
+  TimelineUiTest timeline_ui_test(world_width, &mock_timeline_info, &viewport, &layout);
   timeline_ui_test.TestUpdatePrimitives(0, 100);
+  timeline_ui_test.TestUpdatePrimitives(1'000'000'000, 1'000'000'100);
+  timeline_ui_test.TestUpdatePrimitives(0, 999'999'000);
+  timeline_ui_test.TestUpdatePrimitives(0, 999'999'999);
+  timeline_ui_test.TestUpdatePrimitives(0, 1'000'000'000);
+
+  // Maximum supported timestamp: 1 Month.
```

Comment:
pers pref: Maybe `tested` instead of `supported`. Supported sounds a bit like there will be problems when its more than a month.